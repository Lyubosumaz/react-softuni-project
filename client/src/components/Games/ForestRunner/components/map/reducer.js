import { generalActions } from './actions';

const initialState = {
    tiles: [],
};

const mapReducer = (state = initialState, action) => {
    switch (action.type) {
        case generalActions.SET_TILES:
            return {
                ...action.payload,
            };
        default:
            return state;
    }
};

export default mapReducer;
