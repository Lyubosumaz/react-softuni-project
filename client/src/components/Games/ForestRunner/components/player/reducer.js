const initialState = {
    position: [0, 0],
    spriteLocation: '0px 0px',
    default: 'EAST',
    walkIndex: 0,
};

const playerReducer = (state = initialState, action) => {
    switch (action.type) {
        case 'MOVE_PLAYER':
            return {
                ...state,
                ...action.payload,
            }
        case 'RESET_PLAYER_LOCATION':
            return {
                ...state,
                ...initialState
            }
        default:
            return state;
    };
};

export default playerReducer;