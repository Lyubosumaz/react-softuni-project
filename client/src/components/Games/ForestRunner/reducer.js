const initialState = {
    inGame: false,
    gold: 0,
    item: [],
    time: 0,
    handleGameTimer: false,
    level: 1,
    gameItems: [],
    inventorySellItem: false,
    inventoryEquipItem: false,
    characterRemoveItem: false,
};

const gameReducer = (state = initialState, action) => {
    switch (action.type) {
        case 'SAVE_GAME_ITEMS_LOOT':
            return {
                ...state,
                gameItems: action.payload,
            };
        case 'GAME_TIMER_STATE':
            return {
                ...state,
                handleGameTimer: action.payload,
            };
        case 'SET_TIME':
            return {
                ...state,
                time: action.payload,
            };
        case 'OPEN_GOLD_CHEST':
            return {
                ...state,
                gold: (state.gold += action.payload),
            };
        case 'OPEN_ITEM_CHEST':
            return {
                ...state,
                item: state.item.concat(action.payload),
            };
        case 'START_THE_GAME':
            return {
                ...state,
                inGame: action.payload,
            };
        case 'END_THE_GAME':
            return {
                ...state,
                inGame: action.payload,
                level: 2,
            };
        case 'RESET_GAME_LEVEL':
            return {
                ...state,
                gold: 0,
                item: [],
                time: 0,
            };
        case 'CHARACTER_SELL_ITEM':
            return {
                ...state,
                inventorySellItem: action.payload,
            };
        case 'CHARACTER_EQUIP_ITEM':
            return {
                ...state,
                inventoryEquipItem: action.payload,
            };
        case 'CHARACTER_REMOVE_ITEM':
            return {
                ...state,
                characterRemoveItem: action.payload,
            };
        default:
            return state;
    }
};

export default gameReducer;
