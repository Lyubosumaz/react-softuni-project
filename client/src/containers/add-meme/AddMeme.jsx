import React, { useState } from 'react';
import { connect } from 'react-redux';

import history from '../../services/history';
import schema from './add-meme-validations';

function AddMeme(props) {
    const title = useFormInput('');
    const imageUrl = useFormInput('');
    const [errors, setErrors] = useState({});

    const handleRoute = (name) => (e) => {
        e.preventDefault();
        history.push(name);
    };

    return (
        <div>
            <form>
                <div className="form-container">
                    <h1>Add Meme</h1>
                    <p>Adding meme was never been easier.</p>

                    <div className="form-div-container">
                        <label htmlFor="Title"><b>Title:</b></label>
                        <input type="text" placeholder="Write some funny title" className="form-input" id="title" onChange={useFormInput} {...title} />
                        {errors.title && <div className="form-input-error">{errors.title[0]}</div>}
                    </div>

                    <div className="form-div-container">
                        <label htmlFor="imageUrl"><b>imageUrl:</b></label>
                        <input type="text" placeholder="Copy and Paste your memeURL here" className="form-input" id="imageUrl" onChange={useFormInput} {...imageUrl} />
                        {errors.imageUrl && <div className="form-input-error">{errors.imageUrl[0]}</div>}
                    </div>

                    <div>
                        <button type="submit" className="form-action-btn" onClick={handleSubmit}>Submit</button>
                    </div>

                    <div className="form-info-container">
                        <p>Tired of memes? Start a new game <button className="form-info-button"
                            onClick={handleRoute('/game')}>Here</button>.</p>
                    </div>
                </div>
            </form>
        </div>
    );

    function handleSubmit(e) {
        e.preventDefault();
        const data = {
            title,
            imageUrl,
        };
        const hasErrors = Object.keys(errors).filter(key => errors[key].length > 0);

        if (hasErrors.length === 0 && data.title && data.imageUrl) {
            console.log(props.login.userId)
            console.log(data)
        }
    };

    function useFormInput(initialValue) {
        const [value, setValue] = useState(initialValue);

        function handleChange(event) {
            setValue(event.target.value);
            validate(event);
        };

        function validate(event) {
            const name = event.target.id;

            schema.fields[name].validate(event.target.value, { abortEarly: false })
                .then(() => {
                    setErrors({ ...errors, [name]: [] });
                })
                .catch((err) => {
                    setErrors({ ...errors, [name]: err.errors });
                });
        };

        return { value, onChange: handleChange };
    };
};

function mapStateToProps(state) {
    return {
        ...state
    };
};

export default connect(mapStateToProps)(AddMeme);