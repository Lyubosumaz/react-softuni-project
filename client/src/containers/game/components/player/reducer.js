const initialState = {
    inGame: false,
    position: [0, 0],
    spriteLocation: '0px 0px',
    default: 'EAST',
    walkIndex: 0,
    final: false,
    gold: 0,
    items: []
};

const playerReducer = (state = initialState, action) => {
    switch (action.type) {
        case 'MOVE_PLAYER':
            return {
                ...state,
                ...action.payload,
            }
        case 'OPEN_CHEST':
            return {
                ...state,
                gold: state.gold += action.payload,
            };
        case 'FINAL':
            return;
        case 'START':
            return {
                ...state,
                inGame: action.payload,
            }
        default:
            return state;
    };
};

export default playerReducer;