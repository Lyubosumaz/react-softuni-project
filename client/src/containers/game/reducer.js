const initialState = {
    inGame: false,
    final: false,
    gold: 0,
    item: [],
    time: 0,
    level: 1,
    gameItems: [],
    inventorySellItem: false,
    inventoryEquipItem: false,
    characterRemoveItem: false
};

const gameReducer = (state = initialState, action) => {
    switch (action.type) {
        case 'SAVE_GAME_ITEMS':
            return {
                ...state,
                gameItems: action.payload,
            }
        case 'START':
            return {
                ...state,
                inGame: action.payload,
            }
        case 'OPEN_GOLD_CHEST':
            return {
                ...state,
                gold: state.gold += action.payload,
            };
        case 'OPEN_ITEM_CHEST':
            return {
                ...state,
                item: state.item.concat(action.payload),
            };
        case 'GET_TIME':
            return {
                ...state,
                time: action.payload,
            };
        case 'FINAL':
            return {
                ...state,
                inGame: action.payload,
                level: 2,
            };
        case 'RESET_GAME':
            return {
                ...state,
                final: false,
                gold: 0,
                item: [],
                time: 0,
                // level: 1
            };
        case 'CHARACTER_SELL_ITEM':
            return {
                ...state,
                inventorySellItem: action.payload
            };
        case 'CHARACTER_EQUIP_ITEM':
            return {
                ...state,
                inventoryEquipItem: action.payload
            };
        case 'CHARACTER_REMOVE_ITEM':
            return {
                ...state,
                characterRemoveItem: action.payload
            };
        default:
            return state;
    };
};

export default gameReducer;