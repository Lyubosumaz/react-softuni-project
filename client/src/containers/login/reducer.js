const initialState = {
    isLogin: false,
    userId: null,
}

const loginReducer = (state = initialState, action) => {
    switch (action.type) {
        case 'USER_LOGGED_IN':
            return {
                ...state,
                isLogin: true,
                userId: action.payload._id
            }
        default:
            return state;
    }
}

export default loginReducer;