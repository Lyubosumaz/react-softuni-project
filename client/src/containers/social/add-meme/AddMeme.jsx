import { useState } from 'react';
import history from '../../../utils/history';
import { connect } from 'react-redux';
import http from '../../../services/http';
import { toast } from 'react-toastify';
import schema from './add-meme-validations';
import Title from '../../../components/title/Title';
import Info from '../../../components/info/Info';

function AddMeme(props) {
    const title = useFormInput('');
    const imageUrl = useFormInput('');
    const [errors, setErrors] = useState({});

    function handleSubmit(e) {
        e.preventDefault();
        const meme = {
            title: title.value,
            imageUrl: imageUrl.value,
        };
        const hasErrors = Object.keys(errors).filter((key) => errors[key].length > 0);

        if (hasErrors.length === 0 && meme.title && meme.imageUrl && props.isLogin) {
            http.Social.addMeme(meme)
                .then((res) => {
                    toast(res.message, {
                        type: toast.TYPE.SUCCESS,
                    });
                    history.push('/social');
                })
                .catch((err) => {
                    toast(err.message, {
                        type: toast.TYPE.ERROR,
                    });
                });
        }
    }

    function useFormInput(initialValue) {
        const [value, setValue] = useState(initialValue);

        function handleChange(event) {
            setValue(event.target.value);
            validate(event);
        }

        function validate(event) {
            const name = event.target.id;

            schema.fields[name]
                .validate(event.target.value, { abortEarly: false })
                .then(() => {
                    setErrors({ ...errors, [name]: [] });
                })
                .catch((err) => {
                    setErrors({ ...errors, [name]: err.errors });
                });
        }

        return { value, onChange: handleChange };
    }

    return (
        <div className="add-meme-container">
            <form>
                <Title type={'component'} />

                <div className="form-div-container">
                    <label htmlFor="Title">
                        <b>Title:</b>
                    </label>
                    <input type="text" placeholder="Write some funny title" className="form-input" id="title" onChange={useFormInput} {...title} />
                    {errors.title && <div className="form-input-error">{errors.title[0]}</div>}
                </div>

                <div className="form-div-container">
                    <label htmlFor="imageUrl">
                        <b>imageUrl:</b>
                    </label>
                    <input type="text" placeholder="Copy and Paste your memeURL here" className="form-input" id="imageUrl" onChange={useFormInput} {...imageUrl} />
                    {errors.imageUrl && <div className="form-input-error">{errors.imageUrl[0]}</div>}
                </div>

                <div>
                    <button type="submit" className="form-action-btn" onClick={handleSubmit}>
                        Submit
                    </button>
                </div>

                <Info />
            </form>
        </div>
    );
}

function mapStateToProps(state) {
    return {
        isLogin: state.user.isLogin,
    };
}

export default connect(mapStateToProps)(AddMeme);
