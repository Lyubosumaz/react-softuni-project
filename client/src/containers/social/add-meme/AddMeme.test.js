import { Provider } from 'react-redux';
import { createStore } from 'redux';
import { mount } from 'enzyme';
import reducer from '../../login/reducer';
import AddMeme from './AddMeme';

describe('AddMeme Component', () => {
    let getWrapper;
    beforeEach(() => {
        const mockStore = createStore(reducer, { user: { isLogin: true } });
        jest.mock('history', () => ({
            push: jest.fn(fn => fn()),
        }));
        getWrapper = () => mount(
            <Provider store={mockStore}>
                <AddMeme />
            </Provider>
        );
    });

    it('should renter AddMeme Component', () => {
        const wrapper = getWrapper();
        expect(wrapper).toBeTruthy();
    });

    it('should render without errors main-container', () => {
        const wrapper = getWrapper().find('.main-container');
        expect(wrapper.length).toBe(1);
    });

    it('should checks for html changes', () => {
        const wrapper = getWrapper();
        expect(wrapper).toMatchSnapshot();
    });
});
