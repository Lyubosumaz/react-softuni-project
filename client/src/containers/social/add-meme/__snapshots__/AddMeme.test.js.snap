// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddMeme Component Checks for html changes 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(AddMeme)>
    <AddMeme
      dispatch={[Function]}
      isLogin={true}
    >
      <div
        className="main-container"
      >
        <form>
          <h1>
            Add Meme
          </h1>
          <p>
            Adding meme was never been easier.
          </p>
          <div
            className="form-div-container"
          >
            <label
              htmlFor="Title"
            >
              <b>
                Title:
              </b>
            </label>
            <input
              className="form-input"
              id="title"
              onChange={[Function]}
              placeholder="Write some funny title"
              type="text"
              value=""
            />
          </div>
          <div
            className="form-div-container"
          >
            <label
              htmlFor="imageUrl"
            >
              <b>
                imageUrl:
              </b>
            </label>
            <input
              className="form-input"
              id="imageUrl"
              onChange={[Function]}
              placeholder="Copy and Paste your memeURL here"
              type="text"
              value=""
            />
          </div>
          <div>
            <button
              className="form-action-btn"
              onClick={[Function]}
              type="submit"
            >
              Submit
            </button>
          </div>
          <div
            className="info-container"
          >
            <p>
              You could add meme later, maybe play a 
              <button
                className="info-button"
                onClick={[Function]}
              >
                Game
              </button>
              ?
            </p>
          </div>
        </form>
      </div>
    </AddMeme>
  </Connect(AddMeme)>
</Provider>
`;
